src/data/polynomial/degree/erase_leading.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.degree.basic
+import data.polynomial.degree.to_basic
+import data.polynomial.degree.to_trailing_degree
+import data.polynomial.degree.trailing_degree
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+namespace erase_leading
+
+/-- erase_lead of a polynomial f is the polynomial obtained by
+subtracting from f the leading term of f. -/
+def erase_lead (f : polynomial R) : polynomial R := ⟨ f.support \ singleton f.nat_degree , λ a : ℕ , ite (a = f.nat_degree) 0 f.coeff a , λ a , begin
+  simp only [mem_sdiff, mem_support_iff, ne.def, mem_singleton],
+  split_ifs,
+    { simp only [not_and, pi.zero_apply, not_not, eq_self_iff_true, not_true, iff_false],
+      intros a_1,
+      assumption, },
+    { split,
+        { rw and_imp,
+          intros a1 a2,
+          assumption, },
+        { intros a1,
+          exact ⟨ a1 , h ⟩, }, },
+end ⟩
+
+lemma support (f : polynomial R) :
+ (erase_lead f).support = f.support \ singleton f.nat_degree :=
+rfl
+
+@[simp] lemma coeff_nat_degree : (erase_lead f).coeff f.nat_degree = 0 :=
+begin
+  unfold erase_lead,
+  simp only [coeff_mk, if_true, pi.zero_apply, eq_self_iff_true],
+end
+
+@[simp] lemma coeff_eq_coeff_of_ne {a : ℕ} (h : a ≠ f.nat_degree) : (erase_lead f).coeff a = f.coeff a :=
+begin
+  unfold erase_lead,
+  rw coeff_mk,
+  split_ifs,
+    { exfalso,
+      solve_by_elim, },
+    { refl, },
+end
+
+lemma sum_leading_C_mul_X_pow (f : polynomial R) : f = (erase_lead f) + (C f.leading_coeff) * X^f.nat_degree :=
+begin
+  ext1,
+  by_cases nm : n = f.nat_degree,
+    { subst nm,
+      rw [coeff_add, coeff_C_mul, coeff_X_pow_self, mul_one, coeff_nat_degree, zero_add],
+      refl, },
+    { simp only [*, coeff_eq_coeff_of_ne nm, coeff_X_pow f.nat_degree n, add_zero, coeff_C_mul, coeff_add, if_false, mul_zero], },
+end
+
+lemma nonzero_of_large_support (f0 : 2 ≤ f.support.card) : (erase_lead f).support.nonempty :=
```

Comment:
The name could match the meaning more literally. Either
```suggestion
lemma nonzero_of_large_support (f0 : 2 ≤ f.support.card) : (erase_lead f) ≠ 0 :=
```
or
```suggestion
lemma nonempty_of_large_support (f0 : 2 ≤ f.support.card) : (erase_lead f).support.nonempty :=
```
makes more sense to me.

Reply:
Thanks for the suggestion!  I chose to keep the name nonzero and proved (erase_lead f) ≠ 0.